concordium-consensus/src/Concordium/TimeMonad.hs

```diff
@@ -22,6 +22,17 @@ class Monad m => TimeMonad m where
     default currentTime :: MonadIO m => m UTCTime
     currentTime = liftIO getCurrentTime
 
+    -- |Measure the time of a given action m a.
+    -- Note. One has to remember to factor in laziness when
+    -- reasoning about the returned 'NominalDiffTime'.
+    measureTime :: m a -> m (NominalDiffTime, a)
+    default measureTime :: m a -> m (NominalDiffTime, a)
+    measureTime a = do
+        now <- currentTime
+        res <- a
```

Comment:
Why is this not strict?

Reply:
My intention is, that it is up to the caller to _force_ as much as the caller wants/do not want.